### PR TITLE
add Contributing Guidelines

### PR DESCRIPTION
**Pull request by elff1**
add CONTRIBUTING.md, CODE_OF_CONDUCT.md, issue templates, PR template

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+## Solve an Issue
+If you want to solve an unassigned issue in [#help-wanted](https://github.com/verygoodgraphics/vgg_runtime/labels/help%20wanted), you can reply under the comment thread, waiting for an assignment. You can create a [pull request](#pull-request) when the issue is sovled.
+
+## Pull Request
+If you can fix any bug or make any improvement, welcome to send a pull request. Both core team members and external contributors send pull requests which go through the same review process. Checkout [GitHub's pull request tutorial](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests) in case of any issue.
+
+When opening a pull request, you will be asked to sign a Contributor License Agreement (CLA).
+
+### basic guildlines
```
Comment (harry75369) [inline CONTRIBUTING.md:24]: typo. `Basic Guidelines`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+- Keep each commit atomic, and commit with clear messages.
+- Follow the PR template, and write clear descriptions.
+
+### extra guildines for code
```
Comment (harry75369) [inline CONTRIBUTING.md:28]: typo & grammar problem. `Extra Guidelines for coding`

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+- How to build the code, refer to [Build Instructions](README.md#build-instructions).
+- Make sure you put useful comments in your code.
+- For C++ code, we enforce the coding style via clang-format and clang-tidy.
+
+## Contact Us
+- twitter: https://twitter.com/VGG_Design
```
Comment (harry75369) [inline CONTRIBUTING.md:34]: `Twitter`

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+- Discord: https://discord.gg/g3HJuKP54D
+- e-mail: bd@verygoodgraphics.com
```
Comment (harry75369) [inline CONTRIBUTING.md:36]: `E-mail`

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
+# Contributing Guidelines
+
+First, thank you for using VGG! We welcome for issues, pull requests, or asking/answering questions on [Discard channel](#contact-us). Before that, please read [Code of Conduct](CODE_OF_CONDUCT.md).
```
Comment (harry75369) [inline CONTRIBUTING.md:3]: grammar problem. `... on Discord channel` should be `... in Discord channel`

File: CONTRIBUTING.md (hunk 1 above)
Comment (elff1) [reply CONTRIBUTING.md:24]: fixed

File: CONTRIBUTING.md (hunk 2 above)
Comment (elff1) [reply CONTRIBUTING.md:28]: fixed

File: CONTRIBUTING.md (hunk 4 above)
Comment (elff1) [reply CONTRIBUTING.md:36]: fixed

File: CONTRIBUTING.md (hunk 3 above)
Comment (elff1) [reply CONTRIBUTING.md:34]: fixed

File: CONTRIBUTING.md (hunk 5 above)
Comment (elff1) [reply CONTRIBUTING.md:3]: rejected

**Pull request merged**